File: Codewars/JavaScript/toUnderscore.js
/*
  https://www.codewars.com/kata/529b418d533b76924600085d/train/javascript
*/

function toUnderscore(string) {
  let str = string.toString()
  if (str.length === 1) return str
  return str
    .split("")
    .map((e) => {
      if (e.toLowerCase() === e.toUpperCase()) return e;
      else if (e === e.toUpperCase()) return "_" + e.toLowerCase();
      else return e;
    })
    .join("")
    .slice(1);
}

File: Codewars/Python/to_underscore.py
# https://www.codewars.com/kata/529b418d533b76924600085d/train/python

def to_underscore(string):
    input_str = str(string)
    output = [input_str[0].lower()]
    for c in input_str[1:]:
        if c in 'ABCDEFGHIJKLMNOPQRSTUVWXYZ':
            output.append('_')
            output.append(c.lower())
        else:
            output.append(c)
    return str.join('', output)
